framework/packages/abstract-core/src/native/ibc_client.rs

```diff
@@ -259,20 +265,19 @@ mod tests {
             id: callback_id,
             msg: Some(callback_msg),
             result,
+            sender: AccountId::local(1),
         };
 
-        let actual: CosmosMsg<Empty> = response_msg.into_cosmos_msg(receiver.clone()).unwrap();
-
-        assert_that!(actual).matches(|e| {
-            matches!(
-                e,
-                CosmosMsg::Wasm(cosmwasm_std::WasmMsg::Execute {
-                    contract_addr: _receiver,
-                    // we can't test the message because the fields in it are private
-                    msg: _,
-                    funds: _
-                })
-            )
-        });
+        let actual: CosmosMsg<Empty> = response_msg
+            .clone()
+            .into_cosmos_msg(receiver.clone())
+            .unwrap();
+
+        assert_that!(actual).is_equal_to(CosmosMsg::Wasm(cosmwasm_std::WasmMsg::Execute {
+            contract_addr: receiver,
+            // we can't test the message because the fields in it are private
+            msg: to_json_binary(&IbcCallbackMsg::IbcCallback(response_msg)).unwrap(),
+            funds: vec![],
+        }))
```

Comment:
Could you add a test that would fail before this change?

Reply:
And here it is ! 
https://github.com/AbstractSDK/abstract/pull/277/files#diff-c988e60c4037312258209e3bf737bff66c38698ab415e4fb4027cc89597df296R282